cranelift/jit/src/backend.rs

```diff
@@ -140,7 +140,7 @@ impl JITBuilder {
     /// symbol table. Symbol lookup fn's are called in reverse of the order in which they were added.
     pub fn symbol_lookup_fn(
         &mut self,
-        symbol_lookup_fn: Box<dyn Fn(&str) -> Option<*const u8>>,
+        symbol_lookup_fn: Box<dyn Fn(&str) -> Option<*const u8> + Send>,
```

Comment:
Seeing that `libcall_names` already originally required `Send + Sync` in `JITBuilder`, I wonder if it would be better to also add `Sync` to this.

Reply:
I don't think there is much point in making it `Sync` too as `JITModule` can't be `Sync` anyway. For `libcall_names` it makes sense for it to be `Sync` to be able to pass `cranelift_module::default_libcall_names()`.